File: library/foundation/java/src/library/foundation/Bridge.java
package src.library.foundation;

import src.library.bridge.BridgeValue;

public class Bridge {

    public static long info(long message, long file, long line, long arg3) {
        String aa = BridgeValue.getString(message);
        String bb = BridgeValue.getString(file);
        int cc = BridgeValue.getInt(line);
        Logger.info(aa, bb, cc);
        return BridgeValue.Null;
    }

    public static long error(long message, long file, long line, long arg3) {
        String aa = BridgeValue.getString(message);
        String bb = BridgeValue.getString(file);
        int cc = BridgeValue.getInt(line);
        Logger.error(aa, bb, cc);
        return BridgeValue.Null;
    }
}

File: library/foundation/libfoundation/java_src_library_foundation/Bridge.cpp
#include "cqbridge.hh"
#include "cqbridgejni.hh"

CQ_C_LINK void CQ_Logger_info(CQBridgeValue message) {
    CQJavaCallStatic("src/library/foundation/Bridge", "info", message, CQBridgeValueNull, CQBridgeValueNull, CQBridgeValueNull);
}

CQ_C_LINK void CQ_Logger_error(CQBridgeValue message) {
    CQJavaCallStatic("src/library/foundation/Bridge", "error", message, CQBridgeValueNull, CQBridgeValueNull, CQBridgeValueNull);
}
